Sources/NIO/Codec.swift

```diff
@@ -351,6 +359,9 @@ extension ByteToMessageHandler: ChannelInboundHandler {
 
 
     public func handlerRemoved(ctx: ChannelHandlerContext) {
+        assert(self.removalState == .notBeingRemoved || self.removalState == .removalCompleted,
+               "illegal removal state in handlerRemoved: \(self.removalState)")
+        self.removalState = .removalCompleted
         self.processLeftovers(ctx: ctx)
```

Comment:
Do we still want this call? I think it's only useful if the user removes the handler from the body of `decode` using the `ChannelHandlerContext`, and even there it may be a little spurious as the leftovers aren't reachable (the call re-enters immediately). The only other way we can get here without going through the closure in `removeHandler` is if the pipeline is being torn down, and that teardown means delivering the leftovers is both a) of limited use, and b) semantically invalid (the user never asked to remove B2MD from the pipeline, so the arrival of a ByteBuffer will just cause a crash).

Reply:
oops, no